service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/Twin.java

```diff
@@ -166,11 +172,6 @@ public Twin(String deviceId) throws IllegalArgumentException
     {
         this();
 
-        if (Tools.isNullOrEmpty(deviceId))
-        {
-            throw new IllegalArgumentException("deviceId cannot be null or empty.");
```

Comment:
Device ID is allowed to be null, interestingly. Only when querying twins and filtering out the deviceId field, though

Reply:
Deserves a code comment, somewhere. Thoughts?